packages/extension/src/ui/features/accountTokens/SendTokenScreen.tsx

```diff
@@ -344,30 +346,51 @@ export const SendTokenScreen: FC = () => {
 
   const showSaveAddressButton = validRecipientAddress && !recipientInAddressBook
 
+  const parseInputAmount = (amount?: string, decimals?: number) => {
+    if (amount) {
+      try {
+        const result = parseAmount(amount, decimals ?? 18)
+        setIsFractionalComponentExceedsDecimals(false)
+        return result
+      } catch (e: any) {
+        if (e.message.startsWith('fractional component exceeds decimals')) {
+          setIsFractionalComponentExceedsDecimals(true)
+          console.error("e", e.mesage)
+        } else {
+          throw e
```

Comment:
Maybe introduce `setFormError` so that we could catch this error too?
We could use `setFormError("Insufficient balance"), setFormError("...")`